checkout/models.py

```diff
@@ -0,0 +1,94 @@
+import uuid
+
+from django.db import models
+from django.db.models import Sum
+from django.conf import settings
+
+from products.models import Product
+from profiles.models import UserProfile
+
+
+class Order(models.Model):
+
+	order_number = models.CharField(max_length=32, null=False, editable=False)
+	user_profile = models.ForeignKey(UserProfile, on_delete=models.CASCADE, related_name="orders", null=True, blank=True)
+	full_name = models.CharField(max_length=50, null=False, blank=False)
+	email = models.EmailField(max_length=254, null=False, blank=False, default='')
+	phone_number = models.CharField(max_length=20, null=False, blank=False)
+	country = models.CharField(max_length=40, null=False, blank=False)
+	postcode = models.CharField(max_length=20, null=True, blank=True)
+	town_or_city = models.CharField(max_length=40, null=False, blank=False)
+	street_address1 = models.CharField(max_length=80, null=False, blank=False)
+	street_address2 = models.CharField(max_length=80, null=True, blank=True)
+	county = models.CharField(max_length=80, null=True, blank=True)
+	date = models.DateTimeField(auto_now_add=True)
+	shipping_cost = models.DecimalField(max_digits=6, decimal_places=2, null=False, default=0)
+	order_total = models.DecimalField(max_digits=10, decimal_places=2, null=False, default=0)
+	grand_total = models.DecimalField(max_digits=10, decimal_places=2, null=False, default=0)
+
+
+	def generate_order_number(self):
```

Comment:
If this is only needed as part of the `save` method, I'd make this a private method (preceded by an underscore), or (probably even better), move it outside of the class

Reply:
I agree on making it private. Why would moving it outside be better?